src/snowflake/snowpark/types.py

```diff
@@ -455,6 +479,11 @@ def get_snowflake_col_datatypes(self):
 #: The type hint for annotating Geometry data when registering UDFs.
 Geometry = TypeVar("Geometry")
 
+# TODO(SNOW-969479): Add a type hint that can be used to annotate Vector data. Python does not
+# currently support integer type parameters (which are needed to represent a vector's dimension).
+# typing.Annotate can be used as a temporary bypass once the minimum supported Python version is
+# bumped to 3.9
```

Comment:
Looks like you're not the first person to reach for dependent types in Python: https://typing-extensions.readthedocs.io/en/latest/#Literal 😄

Reply:
Yeah, if `Literal` supported unbounded domains, I think it'd be all we need.